lib/rom/http/relation.rb

```diff
@@ -1,40 +1,34 @@
-require 'dry/core/cache'
 require 'rom/initializer'
-require 'rom/http/transformer'
+
+require 'rom/http/attribute'
+require 'rom/http/schema/dsl'
+require 'rom/http/schema/dsl'
 
 module ROM
   module HTTP
     # HTTP-specific relation extensions
     #
     class Relation < ROM::Relation
-      extend Dry::Core::Cache
-      extend ::ROM::Initializer
-      include Enumerable
-
       adapter :http
 
-      option :transformer, reader: true, default: proc { ::ROM::HTTP::Transformer }
+      schema_class HTTP::Schema
+      schema_dsl HTTP::Schema::DSL
+      schema_attr_class HTTP::Attribute
 
       forward :with_headers, :add_header, :with_options,
               :with_base_path, :with_path, :append_path,
               :with_request_method, :with_params, :add_params
 
       def primary_key
-        attribute = schema.find(&:primary_key?)
-
-        if attribute
-          attribute.alias || attribute.name
-        else
-          :id
-        end
+        schema.primary_key[0].name
```

Comment:
I think we can have a missing method here, mb it's a good idea to raise a meaningful exception